doc/reference/javascript_reference.rst

```diff
@@ -0,0 +1,1806 @@
+
+        var FieldPad = ...;
+
+        fieldRegistry.add('pad', FieldPad);
+
+    Note that each value should be a subclass of *AbstractField*
+
+- view registry: this registry contains all JS views known to the web client
+    (and in particular, the view manager).  Each value of this registry should
+    be a subclass of *AbstractView*
+
+- action registry: we keep track of all client actions in this registry.  This
+    is where the action manager look up whenever it needs to create a client
+    action.  In version 11, each value should simply be a subclass of *Widget*.
+    However, in version 12, the values are required to be *AbstractAction*.
+
+
+Communication between widgets
+=============================
+
+There are many ways to communicate between components.
+
+* From a parent to its child:
+    this is a simple case. The parent widget can simply call a method on its
+    child:
+
+    .. code-block:: javascript
+
+        this.someWidget.update(someInfo);
+
+* From a widget to its parent/some ancestor:
+    in this case, the widget's job is simply to notify its environment that
+    something happened.  Since we do not want the widget to have a reference to
+    its parent (this would couple the widget with its parent's implementation),
+    the best way to proceed is usually to trigger an event, which will bubble up
+    the component tree, by using the ``trigger_up`` method:
+
+    .. code-block:: javascript
+
+        this.trigger_up('open_record', { record: record, id: id});
+
+    This event will be triggered on the widget, then will bubble up and be
+    eventually caught by some upstream widget:
+
+    .. code-block:: javascript
+
+        var SomeAncestor = Widget.extend({
+            custom_events: {
+                'open_record': '_onOpenRecord',
+            },
+            _onOpenRecord: function (event) {
+                var record = event.data.record;
+                var id = event.data.id;
+                // do something with the event.
+            },
+        });
+
+* Cross component
+    Cross component communication can be achieved by using a bus.  This is not
+    the preferred form of communication, because it has the disadvantage of
+    making the code harder to maintain.  However, it has the advantage of
+    decoupling the components.  In that case, this is simply done by triggering
+    and listening to events on a bus.  For example:
+
+    .. code-block:: javascript
+
+        // in WidgetA
+        var core = require('web.core');
+
+        var WidgetA = Widget.extend({
+            ...
+            start: function () {
+                core.bus.on('barcode_scanned', this, this._onBarcodeScanned);
+            },
+        });
+
+        // in WidgetB
+        var WidgetB = Widget.extend({
+            ...
+            someFunction: function (barcode) {
+                core.bus.trigger('barcode_scanned', barcode);
+            },
+        });
+
```

Comment:
Maybe explain that new buses can be created if one doesn't want to use the one from core?